engine/script/src/script_table.cpp

```diff
@@ -395,7 +400,7 @@ namespace dmScript
 
                 case LUA_TTABLE:
                 {
-                    size += DoCheckTableSize(L, -1);
+                    size += DoCheckTableSize(L, -1, total_size + size);
```

Comment:
This looks strange.
WIll this not add the current size twice?

It will work, but the calculated size is way larger than it needs to be?

Reply:
The total_size+size that I pass in here is only used when calculating the align.

```
local parent = {
    foo = 1234,
    child = {
         bar = 5678
    }
}
```

Imagine that we're calculating the size of the above table and that the keys are iterated in the order they are defined (not guaranteed, I know):

* We start with a total_size = 0 and size = 0 when we serialize "parent"
* We write the "foo" key and increase size by X
* We calculate the alignment based on total_size+size (0+X) and write value 1234 and increase size by Y
* We write the "child" key and increase size by Z
* We recurse into serialization of the child table. We pass in the total_size+size
    * We start with a total_size = X+Y+Z and size = 0 when we serialize "child"
    * We write the "bar" key and increase size by W
    * We calculate the alignment based on total_size+size (X+Y+Z+W) and write value 5678

The previous version of the code did not take into account X+Y+Z and W when calculating the alignment of for the value of "bar" and instead assumed it was 0+W.

Reply:
I think one of the reasons I was confused was that you did the "total + size".

If instead you initialize to

```
uint32_t size = offset; // instead of total_size
```
Then you wouldn't need to do the "total + size" later on.

Reply:
Also, I'd like to point out that a common way to do a dry run is to support passing a null pointer for the output buffer.
Then the function would ofc have to handle that case, but it usually makes keeping the two functions in sync.
I honestly think I'd prefer that, and that's what I'd recommend.

Reply:
I've added a test to check calculated vs actual size